modules/persistent-storage-csi-vsphere-change-max-snapshot.adoc

```diff
@@ -0,0 +1,127 @@
+.Example output
++
+[source, terminal]
+----
+apiVersion: v1
+data:
+  cloud.conf: |+
+    # Labels with topology values are added dynamically via operator
+    [Global]
+    cluster-id = vsphere-01-cwv8p
+
+    [VirtualCenter "vcenter.devqe.ibmc.devcluster.openshift.com"]
+    insecure-flag           = true
+    datacenters             = DEVQEdatacenter
+    migration-datastore-url = ds:///vmfs/volumes/vsan:527320283a8c3163-2faa6dc5949a3a28/
+
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2024-03-06T09:46:40Z"
+  name: vsphere-csi-config
+  namespace: openshift-cluster-csi-drivers
+  resourceVersion: "126687"
+----
++
+In this example, the global maximum number of snapshots is not configured, so the default value of 3 is applied.
+
+. Change the snapshot limit by running the following command:
++
+* Set *global* snapshot limit:
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"globalMaxSnapshotsPerBlockVolume": 10}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the global limit is being changed to 10 (`globalMaxSnapshotsPerBlockVolume` set to 10).
+
+* Set *Virtual Volume* snapshot limit:
++
+This parameter sets the limit on the Virtual Volumes datastore only. The Virtual Volume maximum snapshot limit overrides the global constraint if set, but defaults to the global limit if it is not set. 
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"granularMaxSnapshotsPerBlockVolumeInVVOL": 5}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the Virtual Volume limit is being changed to 5 (`granularMaxSnapshotsPerBlockVolumeInVVOL` set to 5).
+
+* Set *vSAN* snapshot limit:
++
+This parameter sets the limit on the vSAN datastore only. The vSAN maximum snapshot limit overrides the global constraint if set, but defaults to the global limit if it is not set. You can set a maximum value of 32 under vSAN ESA setup.
++
+[source, terminal]
+----
+$ oc patch clustercsidriver/csi.vsphere.vmware.com --type=merge -p '{"spec":{"driverConfig":{"vSphere":{"granularMaxSnapshotsPerBlockVolumeInVSAN": 7}}}}'
+clustercsidriver.operator.openshift.io/csi.vsphere.vmware.com patched
+----
++
+In this example, the vSAN limit is being changed to 7 (`granularMaxSnapshotsPerBlockVolumeInVSAN` set to 7). 
+
+.Verification
+
+* Verify that any changes you made are reflected in the config map by running the following command:
++
+[source, terminal]
+----
+$ oc -n openshift-cluster-csi-drivers get cm/vsphere-csi-config -o yaml
+----
++
+.Example output
++
+[source, terminal]
+----
+apiVersion: v1
+data:
+  cloud.conf: |+
+    # Labels with topology values are added dynamically via operator
+    [Global]
+    cluster-id = vsphere-01-cwv8p
+
+    [VirtualCenter "vcenter.devqe.ibmc.devcluster.openshift.com"]
```

Comment:
best if we don't put real URI maybe vcenter.openshift.com?
@RomanBednar

Reply:
Done.